### PR TITLE
Fix python class extract func for class variables

### PR DESCRIPTION
**Pull request by teddylear**
@pranavrao145 @ThePrimeagen 
Fix python class extract func for class variables

Main issue/hack is that I had to put in an override for `self` word as I'm not sure of a way to block that from coming in. Thought it was easier to remove in the spot I put it. If there is a better idea for how to remove any and all ideas welcome.

### CODE REVIEW

File: lua/refactoring/refactor/106.lua [hunk 1]
```diff
@@ -328,7 +328,33 @@ local function get_selected_locals(refactor)
...
+                    end
+                end
+            end
+        end
+    end
+
     local local_def_map = utils.node_text_to_set(local_defs)
+    -- HACK: Can't think of a better way to do this right now,
+    -- just removing `self` if captured
+    if refactor.filetype == "python" and is_class then
```
Comment (pranavrao145) [inline lua/refactoring/refactor/106.lua:354]: Should be alright, we can come back later if need be.

File: lua/refactoring/treesitter/treesitter.lua [hunk 2]
```diff
@@ -122,6 +124,13 @@ function TreeSitter:get_local_defs(scope, region)
     return nodes
 end
 
+function TreeSitter:get_class_vars(scope, region)
+    -- TODO: add validate setting
+    local class_var_nodes = self:loop_thru_nodes(scope, self.class_vars)
+    print("vim.inspect(class_var_nodes):", vim.inspect(class_var_nodes))
```
Comment (pranavrao145) [inline lua/refactoring/treesitter/treesitter.lua:130]: Was this an autogenerated print statement?

Comment (pranavrao145) [inline lua/refactoring/treesitter/treesitter.lua:130]: Extraneous print statement?

**Review by pranavrao145 — APPROVED** [review]
LGTM, I can't really think of a better way to do this either. I think I had to do something like this for TypeScript at some point; I think there's bound to be edge cases for certain languages because we support so many. Worst case we can come back and figure out what to do about these language-specific things later on.

File: lua/refactoring/treesitter/treesitter.lua (hunk 2 above)
Comment (teddylear) [reply lua/refactoring/treesitter/treesitter.lua:130]: Whoops, this should be removed. I'll open another to remove.

**Pull request merged**